landbosse/model/CollectionCost.py

```diff
@@ -479,13 +490,59 @@ def calc_total_cable_length(cable, cable_specs, num_full_strings, num_partial_st
+            cable.total_length = 0
+            cable.num_turb_per_cable = 0
+            return 0, 0
+
+        # If num full strings < = 1, find which cable the final turbine is on, and calculate total cable length
+        # (including the len to substation) using that cable.
+        elif num_full_strings <= 1 and num_partial_strings >= 0:  # Essentially a switch for distributed wind
+
+            # if number of turbines is less than total string capacity, find the terminal cable and find total cable len
+            # up till that cable.
+            if total_turbines <= turbines_per_cable[
+                count]:  # If total turbines in project are less than cumulative turbines uptil and including that cable.
+                # total_cable_len = (num_full_strings * cable.array_cable_len + num_partial_strings * (cable.array_cable_len * perc_partial_string) + len_to_substation)
+                terminal_string = cable.turb_sequence - 1  # Flag the cable that is the actual terminal cable
+
+                if (cable.turb_sequence - 1) == 0:  # That is, if cable #1 can hold more turbines than specified by user
+
+                    cable.num_turb_per_cable = total_turbines
+                    cable.array_cable_len = (
+                            (cable.num_turb_per_cable + cable.downstream_connection) * cable.turb_section_length)
+
+                    total_cable_len = ((num_full_strings * cable.array_cable_len) + (
+                                num_partial_strings * cable.array_cable_len)) + len_to_substation
+
+                else:
+
+                    cable.num_turb_per_cable = total_turbines - turbines_per_cable[(count - 1)]
+                    cable.array_cable_len = (
+                            (cable.num_turb_per_cable + cable.downstream_connection) * cable.turb_section_length)
+
+                    total_cable_len = ((num_full_strings * cable.array_cable_len) +
+                                       (num_partial_strings * cable.array_cable_len)) + len_to_substation
+
+                return total_cable_len, terminal_string
+
+            else:
+
+                total_cable_len = num_full_strings * cable.array_cable_len + num_partial_strings * (
+                            cable.array_cable_len * perc_partial_string)
+
+        else:  # Switch for utility scale landbosse
+            if cable.turb_sequence == len(cable_specs):
+                # Only add len_to_substation to the final cable in the string
+                total_cable_len = (num_full_strings * cable.array_cable_len + num_partial_strings * (
+                            cable.array_cable_len * perc_partial_string) + len_to_substation)
+            else:
+                total_cable_len = (num_full_strings * cable.array_cable_len + num_partial_strings * (
+                            cable.array_cable_len * perc_partial_string))
 
-        return total_cable_len  # todo: add to output csv
+        return total_cable_len, 9999
```

Comment:
Why is 9999 being returned here?

Reply:
Hi @eberlea, please see ( https://github.nrel.gov/akey/LandBOSSE-dev/blob/master/distributed-mode.md#collectioncost ) for a run down of changes made to collection cost module to extend it to DW. 

Followed by this explanation: 

Basically, line 494 in CollectionCost.py in issue_117_integrate_DW_collection is an IF condition. This conditional statement is the first line of code in function calc_total_cable_length(). This conditional statement checks to see if the code has reached the terminal cable; if NO, it returns a tuple of value (cable_length, 9999), where the 1st element in the tuple is the value of the calculated cable length, and 2nd element in this tuple is basically a flag (where 9999 is NO, and 0 is YES). 

So the next time time this function is called for the next cable in our project database (represented by 'cable.turb_sequence'), it checks first to see if the terminal cable was reached (based on the implemented logic in the rest of the function). If condition passes (that is, the terminal cable was accounted for in the previous function call), skip the rest of the calculations in the function (return 0,0), and repeat this for every other cable in the database.  

<><><><><><><><><><><><><><><><><><><><><><><><><><><><>

Not the cleanest implementations for sure, mostly due to shortage of time (task for extending landbosse to DW was planned for a single quarter in fy19). Though, given that I have more time planned on DW this year, I've set aside some time in the coming quarters to clean up this implementation.

Reply:
Ok, thanks for clarifying, this is fine for now.